libgpiod-monitor/main.cpp

```diff
@@ -0,0 +1,211 @@
+        } else {
+            // No need to set file descriptor and corresponding event if not controlling GPIO
+            return GPIOD_CTXLESS_EVENT_POLL_RET_TIMEOUT;
+        }
+    };
+
+    auto event_callback = [inLine, &eventTransition](int event, unsigned int lineIn, const struct timespec* timeOut,
+                            void* data) -> int
+    {
+        if (event != GPIOD_CTXLESS_EVENT_RISING_EDGE and event != GPIOD_CTXLESS_EVENT_FALLING_EDGE) {
+            std::cerr << "Unexpected GPIO event=" << event << " received" << std::endl;
+            return GPIOD_CTXLESS_EVENT_CB_RET_ERR;
+        }
+        if (not timeOut) {
+            std::cerr << "Invalid timeout structure passed" << std::endl;
+            return GPIOD_CTXLESS_EVENT_CB_RET_ERR;
+        }
+        std::cout << "EventCallback timestamp: sec" << static_cast<uint64_t>(timeOut->tv_sec) <<
+                     " nanosec=" << static_cast<uint64_t>(timeOut->tv_nsec) << std::endl;
+        if (lineIn != inLine) {
+            std::cerr << "Event received for unwanted GPIO line=" << lineIn << std::endl;
+            return GPIOD_CTXLESS_EVENT_CB_RET_STOP;
+        }
+
+        eventTransition = event == GPIOD_CTXLESS_EVENT_RISING_EDGE ? true : false;
+        return GPIOD_CTXLESS_EVENT_CB_RET_OK;
+    };
+
+    auto event_callback_wrapper = [](int event, unsigned int lineIn, const struct timespec* timeOut, void* data)-> int {
+        return (*static_cast<decltype(event_callback)*>(data))(event, lineIn, timeOut, NULL);
+    };
+
+    auto retVal = gpiod_ctxless_event_monitor(devName.c_str(),
+                                              GPIOD_CTXLESS_EVENT_RISING_EDGE,
+                                              inLine,
+                                              false,
+                                              "LSMF",
+                                              &monitorTimeout,
+                                              poll_callback,
+                                              event_callback_wrapper,
+                                              &event_callback);
+    if (retVal != 0) {
+        std::cerr << "Event monitoring for GPIO Line=" << inLine << " Failed! with error=" << errno << std::endl;
+        return retVal;
+    } else {
+        std::cout << "Callbacks successfully registered, this thread will just wait now for callbacks to take over" << std::endl;
+        while(not endFlag) {
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            if (eventTransition) {
+                std::cout << "GPIO event registered!" << std::endl;
+            }
+        }
+        return 0;
+    }
+}
+
+// libgpiod client driver for demonstrating context less event loop
+int main(int argc, char **argv)
+{
+
+    int optIn = 0;
+    int numDevices = -1;
+    unsigned int numLine = 0;
+
+    // Boilerplate command line arguments processing
+    while((optIn = getopt(argc, argv, "uhn:l:")) != EOF) {
```

Comment:
I did not know `getopt()` before.  I learned this is useful function!  Cool!  Thanks :)